google_geocoder.go

```diff
@@ -116,29 +137,68 @@ func (g *GoogleGeocoder) Geocode(address string) (*Point, error) {
 	return point, nil
 }
 
-func googleGeocodeQueryStr(address string) (string, error) {
-	url_safe_query := url.QueryEscape(address)
-
-	var queryStr = bytes.NewBufferString("")
-	_, err := queryStr.WriteString(fmt.Sprintf("address=%s", url_safe_query))
-	if err != nil {
-		return "", err
-	}
+func googleFormattedRequestStr(params string) (string, error) {
+	queryStr := fmt.Sprintf("%s&%s", "sensor=false", params)
 
 	if GoogleAPIKey != "" {
-		_, err := queryStr.WriteString(fmt.Sprintf("&key=%s", GoogleAPIKey))
+		queryStrBuffer := bytes.NewBufferString(queryStr)
+
+		_, err := queryStrBuffer.WriteString(fmt.Sprintf("&key=%s", GoogleAPIKey))
+		if err != nil {
+			return "", err
+		}
+
+		return queryStrBuffer.String(), nil
+	} else if GoogleClientID != "" && GooglePrivateKey != "" {
+		queryStrBuffer := bytes.NewBufferString(queryStr)
```

Comment:
I wonder if it makes sense to break these out into separate functions for building a URL depending on the authentication schema, be it API key or Business ClientId and PrivateKey.  

Something like:

```
buildGoogleMapsClientSideQuery(params string) (string, error)
buildGoogleMapsForWorkQuery(params string) (string, error)
buildDefaultGoogleMapsQuery() (string)
```